gp-includes/template.php

```diff
@@ -97,8 +97,15 @@ function gp_nav_menu( $location = 'main' ) {
 	$html  = '';
 	$items = gp_nav_menu_items( $location );
 
+	// Get the current URI.
+	global $wp;
+	$gp_route    = isset( $wp->query_vars['gp_route'] ) ? $wp->query_vars['gp_route'] : '';
+	$current_uri = gp_url( $gp_route );
+
 	foreach ( $items as $link => $title ) {
-		$html .= '<a href="' . $link . '">' . $title . '</a>';
+		// Check if the link matches the current URI base, if true, add 'current' class.
+		$class = str_starts_with( $current_uri, $link ) ? 'current' : '';
```

Comment:
`str_starts_with()` is a PHP 8 function and only polyfilled in WordPress 5.9+. You can use `gp_startswith()` instead.